src/as_execution/common.rs

```diff
@@ -49,6 +51,8 @@ pub(crate) fn call_module(
     if cfg!(not(feature = "gas_calibration")) {
         set_remaining_points(&env, ctx, resp.remaining_gas)?;
     }
+
+    interface.decrement_recursion_counter()?;
```

Comment:
doesn't it interact badly with the decrement in the call end interface call ?

Reply:
removed the one in finish_call